packages/js/src/initializers/estimated-reading-time.js

```diff
@@ -89,10 +89,16 @@ export default function initializeEstimatedReadingTime() {
 
 	dispatch( "yoast-seo/editor" ).loadEstimatedReadingTime();
 
+	// For the Elementor and Block editor, debounce the subscribe function, since this fires almost continuously.
+	// If not debounced, the editor would become very slow.
 	if ( window.wpseoScriptData.isElementorEditor === "1" ) {
-		initializeEstimatedReadingTimeElementor();
+		subscribe( () => {
```

Comment:
Can be simplified to `subscribe( debounceElementorEditor )`. Maybe you could also create the debounced function here: `subscribe( debounce( getEstimatedReadingTimeElementor, 1500, { maxWait: 3000 } ) )`, altough this looks quit rough at first glance you can easily see what is going on: On every event call getEstimatedReadingTimeElementor with a debounce of 1500ms.

Reply:
Agree!